not-supported/smbserver.sh

```diff
@@ -1289,6 +1289,102 @@ $MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
 done  
 }
 
+automatically_empty_recycle_bins() {
+    local SUBTITLE="Automatically empty recycle bins"
+    local count
+    local TEST=""
+
+    # Ask for removal
+    if crontab -u root -l | grep -q "$SCRIPTS/recycle-bin-cleanup.sh"
+    then
+        if yesno_box_yes "It seems like automatic recycle bin cleanup is already configured. Do you want to disable it?" "$SUBTITLE"
+        then
+            crontab -u root -l | grep -v "$SCRIPTS/recycle-bin-cleanup.sh" | crontab -u root -
+            rm -rf "$SCRIPTS/recycle-bin-cleanup.sh"
+            msg_box "Automatic recycle bin cleanup was successfully disabled." "$SUBTITLE"
+        fi
+        return
+    fi
+
+    # Ask for installation
+    msg_box "Automatic recycle bin cleanup does clean up all recycle bin folders automatically in the background.
+It gets executed every day and cleans old files in the recycle bin folders that were deleted more than 2 days ago." "$SUBTITLE"
+    if ! yesno_box_yes "Do you want to enable automatic recycle bin cleanup?" "$SUBTITLE"
+    then
+        return
+    fi
+
+    # Adjust some things
+    count=1
+    while [ $count -le $MAX_COUNT ]
+    do
+        CACHE=$(sed -n "/^#SMB$count-start/,/^#SMB$count-end/p" "$SMB_CONF")
+        if [ -n "$CACHE" ]
+        then
+            TEST+="SMB$count"
+            if ! echo "$CACHE" | grep -q 'recycle:touch'
+            then
+                CACHE=$(echo "$CACHE" | sed "/recycle:repository/a \ \ \ \ recycle:touch = true")
+                sed -i "/^#SMB$count-start/,/^#SMB$count-end/d" "$SMB_CONF"
+                echo -e "\n$CACHE" >> "$SMB_CONF"
+            fi
+        fi
+        count=$((count+1))
+    done
+
+    # Return if none created
+    if [ -z "$TEST" ]
+    then
+        msg_box "No SMB-share created. Please create a SMB-share first." "$SUBTITLE"
+        return
+    else
+        systemctl restart smbd
+    fi
+
+    # Execute
+    cat << AUTOMATIC_CLEANUP > "$SCRIPTS/recycle-bin-cleanup.sh"
+#!/bin/bash
+# Secure the file
+chown root:root "$SCRIPTS/recycle-bin-cleanup.sh"
+chmod 700 "$SCRIPTS/recycle-bin-cleanup.sh"
+count=1
+while [ \$count -le $MAX_COUNT ]
+do
+    CACHE=\$(sed -n "/^#SMB\$count-start/,/^#SMB\$count-end/p" "$SMB_CONF")
+    if [ -n "\$CACHE" ]
+    then
+        SMB_PATH=\$(echo "\$CACHE" | grep "path =" | grep -oP '/.*')
+        if [ -d "\$SMB_PATH" ] && [ -d "\$SMB_PATH/.recycle/" ]
+        then
+            find "\$SMB_PATH/.recycle/" -type f -atime +2 -delete
```

Comment:
Does this delete recursively? I've tried the same thing my self, but never got it working.

Reply:
Yes, it does. I've just tested.